# Download Youtube video and audio

In [37]:
#url = "https://www.youtube.com/watch?v=zduSFxRajkE"
url = "https://www.youtube.com/watch?v=VD6xJq8NguY"

In [38]:
#Download Video

from pytubefix import YouTube

yt = YouTube(url, 'WEB_CREATOR')
ys = yt.streams.get_highest_resolution()
ys.download(filename="test.mp4")

'/Users/anantkumar/Documents/REPO_WORK/self/video_to_doc/test.mp4'

In [39]:
# Download Audio
ys = yt.streams.get_audio_only()
ys.download(filename="Audio",mp3=True)

'/Users/anantkumar/Documents/REPO_WORK/self/video_to_doc/Audio.mp3'

# Notes
1. Pytube has an error and it is not being maintained so using pytubefix which is getting updated

# Transcription

In [62]:
import whisper

model = whisper.load_model("large-v3-turbo")
result = model.transcribe("Audio.mp3")
print(result["text"])

/Users/anantkumar/Documents/REPO_WORK/self/video_to_doc/.venv/lib/python3.11/site-packages/whisper/__init__.py:150: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  checkpoint 

 We've found a new planet, home to octillions of the most extreme beings living in the most absurd and deadly hellscape. In absolute darkness, crushed by the weight of mountains, starved of oxygen, cooked alive, bathed in acid, salt or radiation. And yet, they live for thousands, perhaps millions of years. It turns out, this planet is not in space, it's inside the crust of Earth. This is the deep biosphere, and we basically learned that it exists yesterday. Its volume is at least twice as large as all the Earth's oceans, home to more microbes than the rest of the entire planet. Their total biomass is more than 20 times greater than all humans, livestock and animal wildlife. Let's descend into this mad, deadly world where none of the rules we thought are mandatory for life apply. Going deeper and deeper. Deep life seems to be everywhere we look, below the oceans, near volcanoes, beneath the glaciers of Antarctica, under any landscape we can imagine, and anywhere we live. We'll use a spe

In [2]:
result

{'text': " We've found a new planet, home to octillions of the most extreme beings living in the most absurd and deadly hellscape. In absolute darkness, crushed by the weight of mountains, starved of oxygen, cooked alive, bathed in acid, salt or radiation. And yet, they live for thousands, perhaps millions of years. It turns out, this planet is not in space, it's inside the crust of Earth. This is the deep biosphere, and we basically learned that it exists yesterday. Its volume is at least twice as large as all the Earth's oceans, home to more microbes than the rest of the entire planet. Their total biomass is more than 20 times greater than all humans, livestock and animal wildlife. Let's descend into this mad, deadly world where none of the rules we thought are mandatory for life apply. Going deeper and deeper. Deep life seems to be everywhere we look, below the oceans, near volcanoes, beneath the glaciers of Antarctica, under any landscape we can imagine, and anywhere we live. We'll

In [10]:
len(result["segments"][0]["text"])

81

In [64]:
data = ""
for i in range (0, len(result["segments"])):
    data += result["segments"][i]["text"] +" "+ "Start Time: " + str(round(result["segments"][i]['start'],2)) +" "+ "End Time: " + str(round(result["segments"][i]['end'],2)) + "\n"
    
print(data)

 We've found a new planet, home to octillions of the most extreme beings living in the most Start Time: 0.56 End Time: 5.84
 absurd and deadly hellscape. In absolute darkness, crushed by the weight of mountains, starved Start Time: 5.84 End Time: 12.24
 of oxygen, cooked alive, bathed in acid, salt or radiation. And yet, they live for thousands, Start Time: 12.24 End Time: 19.84
 perhaps millions of years. It turns out, this planet is not in space, it's inside the crust of Start Time: 19.84 End Time: 26.72
 Earth. This is the deep biosphere, and we basically learned that it exists yesterday. Its volume is at Start Time: 26.72 End Time: 34.96
 least twice as large as all the Earth's oceans, home to more microbes than the rest of the entire planet. Start Time: 34.96 End Time: 40.56
 Their total biomass is more than 20 times greater than all humans, livestock and animal wildlife. Start Time: 41.2 End Time: 47.84
 Let's descend into this mad, deadly world where none of the rules we thought

# Generating Docs using langchain

In [29]:
from langchain_ollama import OllamaLLM

model = OllamaLLM(model="llama3.1")
prompt = f"""You are an expert in drafting word documents from video transcripts. The documents should contain table of contents and the first heading should be introduction based on the video transcript decide the other headings.
If possible try to includes quotes from the transcript where ever possible.
You have also been provided with time stamps use them as much as possible in your documents as well
Just provide the document go indepth as much as possible
Think step by step

Video Transcript: {data}
Word Document:"""
a = model.invoke(prompt)

In [30]:
print(a)

It seems like the text you provided is a transcript from an audio or video file, rather than a written document.

However, I can try to provide some insights based on the content.

The transcript appears to be discussing the concept of deep biospheres, which are vast ecosystems that exist beneath the surface of our planet. The speaker mentions that scientists believe there could be 10 such deep biospheres in our solar system, hidden beneath a seemingly dead and frozen surface.

The conversation then shifts to a science project crate from KiwiCo, which is a subscription-based service that delivers hands-on projects to young scientists. The speaker discusses the benefits of this type of learning experience and how it can foster an ongoing exploration mindset.

Finally, the transcript mentions a book or resource related to life assistance managers or planning, but I couldn't find any specific information about it.

If you could provide more context or clarify what you need help with, I'd 

**Deep Biospheres and the Universe**

**Table of Contents**

1. Introduction to Deep Biospheres
2. The Large Size of the Deep Biosphere on Earth
3. How Life Survives in the Deep Biosphere Without Light, Oxygen, or Sane Temperatures
4. Possibility of Deep Biospheres Existing Elsewhere in the Universe

**Introduction to Deep Biospheres**

Start Time: 540.08End Time: 545.76

The deep biosphere on Earth is a vast and mysterious ecosystem that covers a significant portion of our planet. It's estimated to be around 10 kilometers thick, spanning from the top of the crust to about 10 kilometers below the surface.

**The Large Size of the Deep Biosphere on Earth**

Start Time: 545.76End Time: 551.84

To put this in perspective, the deep biosphere is much larger than the size of the entire human population that lives above ground. It's an incredible and largely unexplored world that still holds many secrets.

**How Life Survives in the Deep Biosphere Without Light, Oxygen, or Sane Temperatures**

Start Time: 551.84End Time: 557.92

The conditions down there are extreme – no light, very little oxygen, and temperatures that can range from near-freezing to boiling hot. Yet, life finds a way to thrive in this environment.

**Possibility of Deep Biospheres Existing Elsewhere in the Universe**

Start Time: 557.92End Time: 563.84

Some scientists believe that there could be up to 10 deep biospheres existing elsewhere in our solar system. This raises an interesting question – could life exist on other planets or moons with similar conditions?

**KiwiCo Science Project Crate**

Start Time: 580.32End Time: 587.76

The KiwiCo science project crate is a fun and educational way for kids to learn about radio control and engineering principles.

**About KiwiCo**

Start Time: 615.44End Time: 620.24

KiwiCo is an innovative learning platform that offers hands-on projects, crates, and online content to help develop problem-solving skills in areas like science, engineering, and art.

**Order the KiwiCo Learning Club**

Start Time: 659.52End Time: 664.64

Try out the new KiwiCo Clubs program and get 50% off your first month of subscription using the code KURTSGZAKT.

Note: The remaining text appears to be unrelated to the topic of deep biospheres and is likely an error or irrelevant information.


In [54]:
from langchain import PromptTemplate

template = """<|begin_of_text|><|start_header_id|>system<|end_header_id|>You are an expert in drafting word documents from video transcripts. The documents should contain table of contents and the first heading should be introduction based on the video transcript decide the other headings.
If possible try to includes quotes from the transcript where ever possible.
Just provide the document go indepth as much as possible
Think step by step<|eot_id|>

Video Transcript: {transcript}
Word Document: """

prompt_template = PromptTemplate(
    input_variables=["transcript"],
    template=template
)

In [55]:
print(
    prompt_template.format(
        transcript = result["text"]
    )
)

<|begin_of_text|><|start_header_id|>system<|end_header_id|>You are an expert in drafting word documents from video transcripts. The documents should contain table of contents and the first heading should be introduction based on the video transcript decide the other headings.
If possible try to includes quotes from the transcript where ever possible.
Just provide the document go indepth as much as possible
Think step by step<|eot_id|>

Video Transcript:  We've found a new planet, home to octillions of the most extreme beings living in the most absurd and deadly hellscape. In absolute darkness, crushed by the weight of mountains, starved of oxygen, cooked alive, bathed in acid, salt or radiation. And yet, they live for thousands, perhaps millions of years. It turns out, this planet is not in space, it's inside the crust of Earth. This is the deep biosphere, and we basically learned that it exists yesterday. Its volume is at least twice as large as all the Earth's oceans, home to more mi

In [58]:
prompt = """<|begin_of_text|><|start_header_id|>system<|end_header_id|>You are an expert in drafting word documents from video transcripts. The documents should contain table of contents and the first heading should be introduction based on the video transcript decide the other headings.
If possible try to includes quotes from the transcript where ever possible.
Just provide the document go indepth as much as possible
Think step by step<|eot_id|>

Video Transcript:  We've found a new planet, home to octillions of the most extreme beings living in the most absurd and deadly hellscape. In absolute darkness, crushed by the weight of mountains, starved of oxygen, cooked alive, bathed in acid, salt or radiation. And yet, they live for thousands, perhaps millions of years. It turns out, this planet is not in space, it's inside the crust of Earth. This is the deep biosphere, and we basically learned that it exists yesterday. Its volume is at least twice as large as all the Earth's oceans, home to more microbes than the rest of the entire planet. Their total biomass is more than 20 times greater than all humans, livestock and animal wildlife. Let's descend into this mad, deadly world where none of the rules we thought are mandatory for life apply. Going deeper and deeper. Deep life seems to be everywhere we look, below the oceans, near volcanoes, beneath the glaciers of Antarctica, under any landscape we can imagine, and anywhere we live. We'll use a special duck science drill and start our journey in familiar territory, on land, in the soil, where plants grow and animals row. If Earth were an onion, this is the very top of the very top layer. Soil is a lavish four-way partnership of air, water, minerals and organic matter, bathed in endless energy. Life lives in luxury here. Plants exploit this paradise and produce more than 30 times the biomass of all of Earth's animals each year, in a constant cycle of growth and decay. Only a tiny fraction of the biomass is buried deeper in the ground, supplying juicy resources for almost half a billion years. As we dig deeper, most of the air has been squeezed out, and we cross the water table into a zone saturated with groundwater, rich in minerals and some organic matter. Roots from the most ambitious plants reach down here, and the most common inhabitants are scavengers living off decay. This layer can be pretty cold because it's still slowly warming up from the most recent ice age. We reach bedrock, a foundation of solid rock for all the less solid stuff above, home to fractures filled with water. It can be exposed to the surface or buried hundreds of meters below stuff. Here in the dense bedrock, we're in a weird planet inside the planet, the most thrilling zone of the deep biosphere. As we drill further down, temperatures gradually begin to rise, and soon it gets really hot, and the pressure rises. Underneath 400 meters of rock, the pressure is similar to the surface of Venus. We drill faster now, down to 1000 meters, deeper than the Burj Khalifa is tall. It's about 30 degrees Celsius, and there's almost no free oxygen left. We continue and finally stop almost 4 kilometers down, above us pressing solid rock weighing tens of thousands of tons, with pressures as intense as at the bottom of the Mariana Trench. Down here, it's on average 120 degrees Celsius, even hotter if a magma plume is nearby. The heat is a leftover from Earth's formation, and from the decay of radioactive elements like thorium and uranium that shower the crust with a constant wave of radioactivity. To make things worse, some rocks are mixed with extreme amounts of salt. Hell. And yet, life is thriving. Inside rock. If we zoom in, we see that solid rock is not actually solid, it's traversed by cracks, voids and tiny pores. Sandstone, limestone or basalt are so porous that up to 40% of their volume is actually empty space. But even much denser rocks, like granite, can be split open by cracks and fractures. We've found a gigantic planet-spanning system of micro caves. Free real estate, filled with water and hardcore microbes. And these caves are moving. Just like the atmosphere is constantly mixing air to create weather, down here rocks are mixing to create rock weather. Submerged mountain ranges are shifting and ripping, crashing and merging. Continents smash into each other with the energy of millions of nuclear weapons, but as slow as your fingernails grow. Countless tiny and not-so-tiny earthquakes rip open tiny new fissures and passageways, creating new spaces for life and closing others off forever. In this hot, moving pressure cooker, minerals are forged and baked, and organic molecules are created and destroyed. An insane menu for anyone brave enough to try to survive down here. Let's venture into the system of tiny caves and meet some of them. The most extreme living things. We think the doctillions of microbes live down here, and naturally they are pretty hardcore. The doomsday preppers of the underworld. Some have big, bulky genomes living entirely on their own, basically forming their own ecosystem. Like the bacterium Desulforidae ordex viata. It synthesizes its own food by nibbling carbon or sulfur from the rock and turning it into organic substances. If the conditions get too extreme, or if there is no food around, it kills itself to survive by forming an endospore. It divides into a big and a small part and swallows the small part again, forming a cell within a cell. The outer cell then sheds its water and kills itself, leaving the spore to float around, maybe for thousands of years, until it finds a good place to spring to life again. Others like some company, like the archaea, with the clunky name Altyarchium hamiconexum, that have a rare double membrane covered in weird materials that protect them against the extremes. They shoot out nano-sized grappling hooks to tether themselves and seem to live in cracks and fissures filled with water completely devoid of oxygen, harvest carbon dioxide to create biomass and may sort of eat hydrogen. The conditions in the deep biosphere are so harsh that other microbes share the hard work by forming consortia. They knit themselves together in a biofilm, a very thin sticky net that shields them against the extremes. They are miniature cells, often with a small genome, but each good at one thing. One type of microbe eats methane and excretes its electrons. A second type eats these electrons and converts sulfate into sulfite, that's then eaten by a third microbe, and so on. Some eat iron, others use electrons to turn nitrogen or carbon dioxide into biomass. Life down here found ways to use stuff that's poisonous to most animals to make food and energy. But still life is incredibly hard, and resources and energy are super hard to come by. So the most intense strategy for survival down here is to live forever. Like monks who've taken a vow of poverty, deep microbes consume very little and conserve their energy. Their metabolism is up to a million times slower than microbes at the surface. They had a meal when you were born and are still digesting it. For most of their life, they exist in a slow limbo. Some even slowly cannibalize themselves. Until a sudden influx of resources arrives by pure chance, and then they spring into action and reproduce. With this lifestyle, it seems that extreme microbes can live for centuries, maybe even for millions of years. If they're not hunted to death, of course. Because kilometers deep in limestone habitat, there seem to be spaces big enough for multicellular predators. We've found asexual worms, 100 times longer than microbes, hunting and devouring bacteria. It's not clear if they originated down here, or if many earthquakes opened up fractures for water to carry them into the deep. But there are other fierce predators like rotifers or arthropods in the depths, hunting immortal microbes. We wish we could tell you more insane things about life in the deep biosphere, but there's a problem. We don't know that much yet. For one, we can't really get a good look under kilometers of rock. Drilling down could contaminate the samples with microbes from the surface. We've found living ones in deep mines and brought them into the land. But it's pretty hard to simulate the conditions they feel comfy in, in boiling hot water squeezed under mountains, submerged in deadly chemicals, and still see them through a microscope. And the microbes live so slowly, for so long, that nothing might happen. A lot of what we know about them we got from turning them into a slurry, and looking at what their genes could do, like breathing nitrogen or eating methane. We know that the diversity of life in the deep biosphere must be staggering, and that some of the most hardcore and extreme beings live down there. This is a proper frontier of science, super hard to study, and most of what we know we've learned in just the last 20 years. There is so much more undiscovered that could bring us progress in medicine, energy, the climate, and more. Let's end by moving our gaze from the inside to the outside again. Since we now know how extremely large the deep biosphere is, and how life down there survives without light, oxygen, sane temperatures, or not being covered in poison, could there be deep biospheres all over the universe? Maybe all you need is a planet or moon with internal heat or radiation, and a chemical composition that allows microbes to build the parts they need. Some scientists suggest there could be 10 of them in the solar system, hiding beneath a seemingly dead and frozen surface. So as we learn more and more about the life below our feet, we may accidentally learn about life in the universe. You're lucky. You don't need to drill dozens of kilometers through solid rock to discover something mind-blowing. You can just order some inspiration right to your doorstep. This is a science project crate from our sponsor KiwiCo, who deliver little adventures in a box. Let's explore. There's wooden components and bolts and wheels. And transmitters. It's the delivery bod kit that teaches the basics of radio control in a playful and intuitive way. This crate is part of the new KiwiCo Labs Learning Club, a subscription that takes you along on a learning journey for young scientists to be. Consecutive hands-on projects help develop and strengthen problem-solving skills on things like science, engineering, and robots over time. There are also different programs available on topics like world geography or art and design, all with their own high-quality crates and exclusive online content to unlock. We're not going to pretend this time that we got it for our own kids. We just ordered it for fun. But we can attest that it makes a great gift for any niece, nephew, gob-child, or other little one in your life. We like KiwiCo because with their new learning club program, they're fostering a mindset of making learning something that's an ongoing exploration, just like we want to. If you want to try out the new KiwiCo Clubs, click on the link in the info box or use the code KURTSGZAKT to get 50% off your first month of a monthly subscription to KiwiCo. And with to go even more. we are constantly in the information Yeah, thank you charge we just our own hands. How can you test what your aim is?
Word Document: 
"""

In [59]:
print(model(prompt))

It seems like there's been a mistake! The provided text doesn't appear to be related to the questions you asked earlier.

However, I'll provide an answer based on the given text:

It's not clear how one could "test" their aim in the context of studying deep biosphere microbes. The process of researching these microbes is more akin to exploration and discovery rather than a target-oriented activity.

If you'd like to ask a specific question about this topic, I'm here to help!


<|begin_of_text|><|start_header_id|>system<|end_header_id|>You are an expert in drafting word documents from video transcripts. The documents should contain table of contents and the first heading should be introduction based on the video transcript decide the other headings.
If possible try to includes quotes from the transcript where ever possible.
Just provide the document go indepth as much as possible
Think step by step<|eot_id|>

Video Transcript: {transcript}
Word Document:


# Generating Docs using Ollama lib

In [65]:
data

" We've found a new planet, home to octillions of the most extreme beings living in the most Start Time: 0.56 End Time: 5.84\n absurd and deadly hellscape. In absolute darkness, crushed by the weight of mountains, starved Start Time: 5.84 End Time: 12.24\n of oxygen, cooked alive, bathed in acid, salt or radiation. And yet, they live for thousands, Start Time: 12.24 End Time: 19.84\n perhaps millions of years. It turns out, this planet is not in space, it's inside the crust of Start Time: 19.84 End Time: 26.72\n Earth. This is the deep biosphere, and we basically learned that it exists yesterday. Its volume is at Start Time: 26.72 End Time: 34.96\n least twice as large as all the Earth's oceans, home to more microbes than the rest of the entire planet. Start Time: 34.96 End Time: 40.56\n Their total biomass is more than 20 times greater than all humans, livestock and animal wildlife. Start Time: 41.2 End Time: 47.84\n Let's descend into this mad, deadly world where none of the rules we

In [56]:
ollama.pull(' t')

KeyboardInterrupt: 

In [68]:
import ollama

response = ollama.generate(model='llama3:text', prompt='''You are an expert in drafting word documents from video transcripts. The documents should contain table of contents and the first heading should be introduction based on the video transcript decide the other headings.
If possible try to includes quotes from the transcript where ever possible.
You have also been provided with time stamps use them as much as possible in your documents as well
Just provide the document go indepth as much as possible
Think step by step

Video Transcript: {data}
Word Document:''')
                           
print(response)

{'model': 'llama3:text', 'created_at': '2024-10-04T20:35:26.887055Z', 'response': ' {data} (with TOC)\n\n', 'done': True, 'done_reason': 'stop', 'context': [2675, 527, 459, 6335, 304, 59000, 3492, 9477, 505, 2835, 61412, 13, 578, 9477, 1288, 6782, 2007, 315, 8970, 323, 279, 1176, 14836, 1288, 387, 17219, 3196, 389, 279, 2835, 36815, 10491, 279, 1023, 70552, 627, 2746, 3284, 1456, 311, 5764, 17637, 505, 279, 36815, 1405, 3596, 3284, 627, 2675, 617, 1101, 1027, 3984, 449, 892, 50312, 1005, 1124, 439, 1790, 439, 3284, 304, 701, 9477, 439, 1664, 198, 10156, 3493, 279, 2246, 733, 5278, 8658, 439, 1790, 439, 3284, 198, 39787, 3094, 555, 3094, 271, 10955, 91403, 25, 314, 695, 534, 11116, 12051, 25, 314, 695, 92, 320, 4291, 5257, 34, 696], 'total_duration': 2919699958, 'load_duration': 2320167000, 'prompt_eval_count': 96, 'prompt_eval_duration': 296003000, 'eval_count': 9, 'eval_duration': 299924000}


In [82]:
import ollama

response = ollama.generate(
    model='llama3.1',   
    prompt=f'''You are a provided a text.
Text: {data}
Provide only Detailed Summary Word Document of Youtube Video use timestamps if necessary for the user to understand and refer the time: ''' 
)

print(response['response'])


**Detailed Summary Word Document**

**Video Title:** Unlocking the Deep Biosphere: A Frontier of Science

**Timestamps:**

* 505.36 - Start Time
* 510.48 - Description of microorganisms' unique abilities
* 516.0 - Description of the deep biosphere's diversity and extremophiles
* 521.04 - Challenges in studying the deep biosphere
* 527.68 - Importance of continued research
* 535.36 - Transition to discussing extraterrestrial life
* 540.08 - Large size of the deep biosphere
* 545.76 - Survival conditions without light, oxygen, or sane temperatures
* 551.84 - Possibility of deep biospheres on other planets/moons
* 557.92 - Chemical composition and internal heat/radiation as requirements for life
* 563.84 - Hypothesis of 10 deep biospheres in the solar system
* 569.6 - Connection between studying life below our feet and life in the universe
* 576.24 - Conclusion on the importance of continued research

**Summary:**

The video discusses the deep biosphere, a vast and largely unexplored fron

**Summary Report**

**Title:** Exploring the Deep Biosphere with Dr. Charles Cockell

**Timestamps:**

* Start Time: 505.36
* End Time: 580.32 (main content)
* Start Time: 587.76 ( KiwiCo commercial break)

**Detailed Summary:**

The video features Dr. Charles Cockell, a scientist who has spent his career studying the deep biosphere. He begins by discussing the vastness of life in the deep Earth and the extreme conditions that living organisms can withstand.

* **506.16**: Dr. Cockell mentions that scientists have discovered microorganisms that can breathe nitrogen or eat methane.
* **510.48**: He emphasizes the importance of studying the deep biosphere, highlighting its diversity and extreme conditions.
* **516.0**: The scientist notes that there is still much to be learned about this environment and its potential applications.

The video then takes a philosophical turn as Dr. Cockell explores the possibility of deep biospheres existing elsewhere in the universe.

* **521.04**: He suggests that life on other planets or moons could thrive in environments with internal heat or radiation, allowing microbes to build the parts they need.
* **545.76**: The scientist proposes that there may be 10 deep biospheres in our solar system, hidden beneath frozen surfaces.

The video concludes with a commercial break featuring KiwiCo, a company that delivers science projects and learning materials to children.

**KiwiCo Commercial Break:**

The commercial showcases a KiwiCo science project crate, which includes components and tools for building radio-controlled delivery bots. The crate is part of the new KiwiCo Labs Learning Club subscription service, designed for young scientists to develop problem-solving skills through hands-on projects.

* **594.08**: Dr. Cockell's voiceover continues as he explores the KiwiCo crate.
* **620.24**: He discusses the importance of developing problem-solving skills in children and highlights the various programs available on KiwiCo, including science, engineering, robots, world geography, and art design.

The video ends with a call-to-action, encouraging viewers to try out the new KiwiCo Clubs subscription service using the code KURTSGZAKT for 50% off their first month.

**Timestamps:**

* Start Time: 505.36 (beginning of Dr. Cockell's presentation)
* End Time: 580.32 (main content ends)
* Start Time: 587.76 (KiwiCo commercial break begins)
* End Time: 680.32 (commercial break ends)


**Summary of YouTube Video**

**Timestamp:** 0:00 - 0:30 (Introduction)

The video starts with an introduction, where the host discusses the vastness and diversity of life on Earth, particularly in the deep biosphere. They highlight that there are many extreme and hard-to-study environments on our planet.

**Timestamp:** 0:31 - 1:15 (Deep Biosphere Explorations)

The host explains how scientists have only begun to explore the deep biosphere in the last 20 years, despite its vast size and complexity. They mention that this frontier of science is still largely unexplored and could lead to significant discoveries in medicine, energy, climate change, and more.

**Timestamp:** 1:16 - 2:30 (Deep Biospheres in the Universe)

The host speculates about the possibility of deep biospheres existing on other planets or moons in our solar system. They suggest that if a planet has internal heat or radiation and a chemical composition suitable for microbial life, it could support its own deep biosphere.

**Timestamp:** 2:31 - 3:15 (Implications of Deep Biospheres)

The host notes that studying the deep biosphere on Earth might inadvertently reveal information about life elsewhere in the universe. They highlight the excitement and potential discoveries that come with exploring this uncharted territory.

**Timestamp:** (Sponsor Segment - KiwiCo Science Project Crate)

Note: The sponsor segment has been skipped, as per your request.

**Summary (Continued)**

The host concludes by mentioning a science project crate from their sponsor, KiwiCo. They encourage viewers to try out the new KiwiCo Clubs and use the code KURTSGZAKT for 50% off their first month's subscription.

Please let me know if you'd like me to add anything else!


**Detailed Summary Document**

The video discusses the deep biosphere, a vast and largely unexplored region beneath our feet where life thrives without light, oxygen, or stable temperatures. The host highlights the diversity of life in this environment, including microbes that can breathe nitrogen or eat methane.

**Paragraph 1: Introduction to the Deep Biosphere (Start Time: 510.48 - End Time: 515.36)**

The video begins by introducing the concept of the deep biosphere, a region beneath our feet where life exists without light, oxygen, or stable temperatures. The host notes that this environment is still largely unexplored and that scientists have only recently begun to understand its complexity.

**Paragraph 2: Exploring the Deep Biosphere (Start Time: 516.0 - End Time: 521.04)**

The host explains that the deep biosphere is a vast and complex environment, with some of the most extreme and hardcore beings on Earth living in this region. The video highlights the challenges of studying this environment, which has only been extensively explored in the last 20 years.

**Paragraph 3: Unexplored Regions and Potential Discoveries (Start Time: 521.04 - End Time: 527.68)**

The host notes that despite significant progress in understanding the deep biosphere, there is still much to be discovered. The video suggests that this region could hold secrets to new medicines, energy sources, and climate solutions.

**Paragraph 4: Extending Our Understanding of Life (Start Time: 527.68 - End Time: 534.32)**

The host reflects on the implications of discovering life in the deep biosphere, suggesting that it may be possible for similar environments to exist elsewhere in the universe. The video highlights the potential for deep biospheres to exist on other planets or moons with internal heat or radiation.

**Paragraph 5: Inspiration from Science (Start Time: 535.36 - End Time: 539.2)**

The host shifts focus to a science project crate from KiwiCo, which delivers hands-on learning experiences to young scientists. The video showcases the delivery bod kit, which teaches radio control basics in a playful and intuitive way.

**Paragraph 6: Learning Clubs and Subscription (Start Time: 540.08 - End Time: 545.76)**

The host explains that KiwiCo offers a subscription-based service called KiwiCo Labs Learning Club, which provides consecutive hands-on projects to develop problem-solving skills in science, engineering, and robotics.

**Paragraph 7: Gift Ideas and Mindset (Start Time: 545.76 - End Time: 551.84)**

The host notes that the KiwiCo subscription makes a great gift for young scientists and fosters a mindset of ongoing exploration and learning.

**Paragraph 8: Call to Action (Start Time: 551.84 - End Time: 557.92)**

The video concludes by encouraging viewers to try out the KiwiCo Clubs, with a special offer of 50% off the first month's subscription using the code KURTSGZAKT.

**Paragraph 9: Final Thoughts and Sponsorship (Start Time: 563.84 - End Time: 569.6)**

The host reflects on the potential for discovering life in the universe, highlighting the importance of continued exploration and learning. The video ends with a final thought on the value of hands-on learning experiences like KiwiCo's science project crates.

# Final Small Context code

In [ ]:
from pytubefix import YouTube
import whisper

In [ ]:
def download_video_audio(link):
    """
    This will download the video and audio from youtube using pytubefix since the original pytube is not maintained anymore.
    """

    yt = YouTube(url, 'WEB_CREATOR')
    # ys = yt.streams.get_highest_resolution()
    # ys.download(filename="video.mp4")
    ys = yt.streams.get_audio_only()
    ys.download(filename="Audio",mp3=True)

def generate_transcription():
    model = whisper.load_model("large-v3-turbo")
    result = model.transcribe("Audio.mp3")
    #print(result["text"])
    transcription = ""
    for i in range (0, len(result["segments"])):
        transcription += result["segments"][i]["text"] +" "+ "Start Time: " + str(round(result["segments"][i]['start'],2)) +" "+ "End Time: " + str(round(result["segments"][i]['end'],2)) + "\n"
    return transcription

